src/PIL/Image.py

```diff
@@ -2396,18 +2396,11 @@ def getdata(self):
         :returns: An :py:class:`~PIL.Image.Image` object.
         """
 
-        if self.mode == "LA":
+        if self.mode in ("LA", "RGBA"):
             return (
-                self.convert("La")
+                self.convert(self.mode.replace("A", "a"))
```

Comment:
```suggestion
                self.convert(self.mode[:-1] + "a")
```
To match `resize()`.